web/styles/buttons.css

```diff
@@ -0,0 +1,186 @@
+.action-button {
+    /* Reset button styles — borrowed from TailwindCSS Preflight */
+    font-family: inherit;
+    font-feature-settings: inherit;
+    font-variation-settings: inherit;
+    font-size: 100%;
+    margin: 0;
+    text-transform: none;
+    appearance: button;
+    background-image: none;
+    cursor: pointer;
+
+    display: flex;
+    gap: 0.5ch;
+    justify-content: center;
+    align-items: center;
+    line-height: 1.3333;
+    font-weight: 550;
+    letter-spacing: 0.02ch;
+    padding: 0.2667em 0.6667em;
+    color: var(--color-btn-text, --color-neutral-secondary-btn-text);
+    background-color: var(--color-btn-bg, --color-neutral-secondary-btn-bg);
+    border-radius: 4px;
+    white-space: nowrap;
+    user-select: none;
+    border: none;
+}
+
+.action-button__label {
+    max-width: 32ch;
+    text-overflow: ellipsis;
+    overflow: hidden;
+}
+
+.action-button:hover {
+    background-color: var(
+        --color-btn-bg-hover,
+        --color-neutral-secondary-btn-bg-hover
+    );
+}
+
+.action-button:active {
+    background-color: var(
+        --color-btn-bg-active,
+        --color-neutral-secondary-btn-bg-active
+    );
+    scale: 0.96;
+}
+
+/* Buttons -- Neutral Variant */
+.action-button--primary-neutral {
+    --color-btn-text: var(--color-neutral-primary-btn-text);
```

Comment:
I'm not 100% sold on this pattern of setting more generic CSS variables scoped to classes. A few reasons:

1. It's an extra lookup step in development, as you'll need to look at the `buttons.css` file to then figure out which color is really (supposed to be) in use.
2. That could also make the variables listed in `app_variables.css` harder to track down across the codebase, since presumably the generics here are the variables to be used with specific CSS properties.
3. The one-off `box-shadow` property on certain buttons makes this implementation also seem less clean to me, and might also invite other one-off properties to creep in over time.

Reply:
The argument could also be made that this doesn't really save much. I'd rather have all the non-color styles in an `.action-button` class, and then see the explicit property-variable listings in the individual classes.

Reply:
Yeah I tend to agree that would feel more readable to me.

Reply:
Went ahead and simplified this:
- Property-value pairs are set directly instead of routing via the custom properties.
- The hover & active states are nested inside `app_variables.css` so that it is easier to read.

Now the action button modifier classes look as such:
```css
.action-button-quiet-neutral {
    color: var(--color-text-neutral-quiet-action-button);
    background-color: var(--color-background-neutral-quiet-action-button);
    box-shadow: 0 0 0.5px 0.5px var(--color-inner-shadow-action-button) inset;

    &:hover {
        background-color: var(
            --color-background-neutral-quiet-action-button-hover
        );
    }

    &:active {
        background-color: var(
            --color-background-neutral-quiet-action-button-active
        );
    }
}
```